lib/vrl/stdlib/src/parse_json.rs

```diff
@@ -162,5 +281,41 @@ mod tests {
                 .add_array(Collection::from_unknown(inner_kind()))
                 .add_object(Collection::from_unknown(inner_kind())),
         }
+
+        max_depth {
+            args: func_args![ value: r#"{"top_layer": {"layer_one": "finish", "layer_two": 2}}"#, max_depth: 1],
+            want: Ok(value!({ top_layer: r#"{"layer_one": "finish", "layer_two": 2}"# })),
+            tdef: type_def(),
+        }
+
+        max_depth_array {
+            args: func_args![ value: r#"[{"top_layer": {"next_layer": ["finish"]}}]"#, max_depth: 2],
+            want: Ok(value!([{ top_layer: r#"{"next_layer": ["finish"]}"# }])),
+            tdef: type_def(),
+        }
+
+        max_depth_exceeds_layers {
+            args: func_args![ value: r#"{"top_layer": {"layer_one": "finish", "layer_two": 2}}"#, max_depth: 10],
+            want: Ok(value!({ top_layer: {layer_one: "finish", layer_two: 2} })),
+            tdef: type_def(),
+        }
```

Comment:
I'm looking for a test that shows that `max_depth` affects the output. If I'm reading this right, the depth limit is not changing the output in any of these tests.

Reply:
Never mind, I misread the quoted bits in the output lines. I was expecting to see the inner values dropped and not left as strings.